pkg/bootkube/create.go

```diff
@@ -205,24 +211,175 @@ func customResourceDefinitionKindURI(apiGroup, version, namespace, plural string
 		strings.ToLower(plural))
 }
 
-func apiTest(c clientcmd.ClientConfig) error {
-	config, err := c.ClientConfig()
+func (c *creater) create(m manifest) error {
+	plural, err := c.mapper.resourcePlural(m.apiVersion, m.kind)
 	if err != nil {
-		return err
+		return fmt.Errorf("dicovery failed: %v", err)
 	}
-	client, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return err
+
+	err = c.client.Post().
+		AbsPath(m.urlPath(plural)).
+		Body(m.raw).
+		SetHeader("Content-Type", "application/json").
+		Do().Error()
+	if err != nil && errors.IsAlreadyExists(err) {
+		// retry with a PUT
```

Comment:
I believe the behavior of "do not replace if exists" is assumed in some places (e.g. recover) - so I'm not sure this is a safe assumption to make.